src/strategy_manager/strategy.manager.cpp

```diff
@@ -0,0 +1,146 @@
+/******************************************************************************
+ * Copyright © 2013-2019 The Komodo Platform Developers.                      *
+ *                                                                            *
+ * See the AUTHORS, DEVELOPER-AGREEMENT and LICENSE files at                  *
+ * the top-level directory of this distribution for the individual copyright  *
+ * holder information and the developer policies on copyright and licensing.  *
+ *                                                                            *
+ * Unless otherwise agreed in a custom licensing agreement, no part of the    *
+ * Komodo Platform software, including this file may be copied, modified,     *
+ * propagated or distributed except according to the terms contained in the   *
+ * LICENSE file                                                               *
+ *                                                                            *
+ * Removal or modification of this copyright notice is prohibited.            *
+ *                                                                            *
+ ******************************************************************************/
+
+#include <vector>
+#include <unordered_map>
+
+#include "strategy.manager.hpp"
+
+namespace antara
+{
+
+bool asset::operator==(const asset &other) const
+{
+  return symbol.value() == other.symbol.value();
+};
+
+bool asset::operator!=(const asset &other) const
+{
+  return !(this->operator==(other));
+}
+
+std::pair<asset, asset> pair::to_std_pair()
+{
+  return std::pair<asset, asset>(quote, base);
+};
+
+bool pair::operator==(const pair &rhs) const
+{
+  return base == rhs.base && quote == rhs.quote;
+};
+
+market_making_strategy::market_making_strategy(
+    antara::pair pair,
+    antara::st_spread spread,
+    antara::st_quantity quantity,
+    antara::side side)
+{
+  this->pair = pair;
+  this->spread = spread;
+  this->quantity = quantity;
+  this->side = side;
+};
+
+bool market_making_strategy::operator==(const market_making_strategy &other) const
+{
+  return pair == other.pair
+      && spread == other.spread
+      && quantity == other.quantity
+      && side == side;
+}
+
+strategy_manager::strategy_manager() {}
+
+void antara::strategy_manager::add_strategy(antara::pair pair, market_making_strategy strat)
+{
+  strategies.emplace(pair, strat);
+}
+
+void strategy_manager::add_strategy(market_making_strategy strat)
```

Comment:
you are copying here a strat each time you call it:

you have possibility:

reference: 

```cpp
void strategy_manager::add_strategy(const market_making_strategy& strat)
```